src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs

```diff
@@ -15,9 +18,12 @@ internal partial class TestingPlatformCommand : CliCommand, ICustomHelp
 
         private MSBuildConnectionHandler _msBuildConnectionHandler;
         private TestModulesFilterHandler _testModulesFilterHandler;
+        private TerminalTestReporter _output;
         private TestApplicationActionQueue _actionQueue;
         private Task _namedPipeConnectionLoop;
         private List<string> _args;
+        private Dictionary<TestApplication, (string ModulePath, string TargetFramework, string Architecture, string ExecutionId)> _executions = new();
```

Comment:
Could we have some race here?

Reply:
The incoming events are from message bus (queue), there should not be parallelism in handling them.

Reply:
there is.